src/channel.ts

```diff
@@ -1547,13 +1547,19 @@ export class Channel<
         if (event.message) {
           /* if message belongs to current user, always assume timestamp is changed to filter it out and add again to avoid duplication */
           const ownMessage = event.user?.id === this.getClient().user?.id;
-          s.addMessageSorted(event.message, ownMessage);
+
+          if (this.state.isUpToDate || event.message.parent_id) {
```

Comment:
thread message with `show_in_channel` flag? maybe disable this completely for threaded messages

Reply:
Good catch